public/js/webcam.js

```diff
@@ -66,26 +69,22 @@ function postTracks(dataURL) {
         });
 }
 
-getTracksButton.addEventListener("click", ()=>{
-    console.log("CLICK");
-    const dataURL = canvas.toDataURL();
-    webcam.style.display = "none";
-    loadingDiv.style.display = "flex";
-    postTracks(dataURL);
+getTracksButton.addEventListener("click", () => {
+    const dataURL = canvas.toDataURL('image/png', 1);
+    webcam.style.setProperty('display', "none");
+    loadingDiv.style.removeProperty('display');
+    return postTracks(dataURL);
 })
 
 takePhotoButton.addEventListener("click", () => {
+    // dimension of pixels in canvas
     canvas.width = video.offsetWidth;
```

Comment:
NOTE: 
`canvas.width` and `canvas.height` are for the dimensions of the pixels in the canvas
`canvas.style.width` and `canvas.style.height` are for the display size of the canvas

In webcamCSS by setting `canvas.style.height` to 70vmin, it will have same display size as video element, and even though `canvas.width` and `canvas.height` will round to integers, I think the pixel ratio will still maintain very close to 1:1.

Reply:
Nice catch!